### PR TITLE
Octokit paginate

### PR DESCRIPTION
**Pull request by whilefoo**
This will replace manual for loop pagination with octokit's integrated plugin [paginate](https://github.com/octokit/plugin-paginate-rest.js/)

### CODE REVIEW

File: src/helpers/issue.ts [hunk 1]
```diff
@@ -7,46 +7,18 @@ import {
...
-
-    await checkRateLimitGit(response?.headers);
-
-    if (response.data.length > 0) {
-      events.push(...(response.data as Event[]));
-      page++;
-    } else {
-      shouldFetch = false;
-    }
-  }
```
Comment (pavlovcik) [inline src/helpers/issue.ts:49]: It doesn't look like you replaced the functionality of looping through every page of results.

**Comment by pavlovcik** [comment]
This is great to see.

File: src/helpers/issue.ts [hunk 2]
```diff
@@ -183,104 +159,76 @@ export async function getIssueDescription(
...
-      if (response?.data?.length > 0) {
-        response.data.filter((item) => item.event === "assigned").forEach((item) => result?.push(item as AssignEvent));
-        page_number++;
-      } else {
-        shouldFetch = false;
-      }
-    }
-  } catch (e: unknown) {
-    shouldFetch = false;
+    return events.sort((a, b) => (new Date(a.created_at) > new Date(b.created_at) ? -1 : 1));
```
Comment (0x4007) [inline src/helpers/issue.ts:227]: Perhaps sorting should occur outside of this function? It makes sense to me that the function only does what its name suggests.

File: src/helpers/issue.ts [hunk 3]
```diff
@@ -367,103 +315,108 @@ export async function isUserAdminOrBillingManager(
...
-      await context.event.octokit.rest.issues.deleteLabel({
+      await context.octokit.rest.issues.deleteLabel({
         owner: payload.repository.owner.login,
         repo: payload.repository.name,
         name: label,
       });
     }
   } catch (e: unknown) {
-    context.logger.debug("Deleting label failed!", e);
+    context.logger.error("Deleting label failed!", e);
```
Comment (0x4007) [inline src/helpers/issue.ts:346]: I was thinking of renaming this method to `fatal` and change `warn` to error to be used in most cases.

**Review by 0x4007 — APPROVED** [review]
Looks good. Please remove the unused variables and functions that were caught in the pull request review GitHub Actions annotations.

**Pull request merged**